pandas/core/arrays/numeric.py

```diff
@@ -199,3 +201,48 @@ def reconstruct(x):
             return tuple(reconstruct(x) for x in result)
         else:
             return reconstruct(result)
+
+    def _apply(self, func: Callable, **kwargs) -> NumericArray:
```

Comment:
the return type is the same as self?

Reply:
I return `return type(self)(values, self._mask.copy())` so, yes, it is a `NumericArray`

Reply:
I guess it depends on the func signature and the subclass constructors. If a func say changes dtype and the subclass constructor accepts the changed dtype. Is that what we want from apply? or would we want say an FloatArray if we apply a func with a signature Callable[[np.ndarray[int],...],np.ndarray[float]] to say a IntegerArray?

we may want to be more precise and restrictive on the callable and say that the func must return the same type (for now) and as a follow-up implement a more generic _apply function

Reply:
As I mentioned on the previous PR, I would keep it focused here on just getting it working for `round`. It's true that if we want such a generic `_apply` for several cases, all those questions come up (return type of the function, does the function return a view or new object, etc). But let's deal with those issues when they actually come up. 

Personally I think we can also just put those 2 lines of code in `round()` itself for now, but documenting the current restrictions of `_apply` (func needs to return new object of same dtype) is also fine for me.